compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1725,7 +1725,16 @@ impl<'a> Parser<'a> {
                 err.help("see https://doc.rust-lang.org/book/ch05-03-method-syntax.html for more information");
                 err
             } else {
-                self.expected_ident_found()
+                let mut err = self.expected_ident_found();
+                if self.check_ident() {
+                    self.bump();
+                    let (ident, _) = self.ident_or_err()?;
+                    err.help("the let keyword is not allowed in struct field definitions");
```

Comment:
Surround keywords with backticks, and this could be a structured suggestion

```suggestion
                    err.span_suggestion_verbose(
                        self.prev_token.span.to(ident.span),
                        "the `let` keyword is not allowed in struct field definitions",
                        String::new(),
                        Applicability::MachineApplicable,
                    );
```